### PR TITLE
Unit tests

### PR DESCRIPTION
**Pull request by Honny1**
This PR includes:
- New unit tests for data structures
- FIxes discovered bugs
- Disables CPE graph generating (panned to fix in other PR) 
- Reorganization of tests
- Independent execution of unit tests and integration tests

### CODE REVIEW

File: tests/unit_tests/test_data_structure.py [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+            assert report.get_severity_of_failed_rules_stats()
+
+
+@pytest.mark.parametrize("system, type_of_remediation", [
+    ("Unknown_system", "Unknown_system"),
+    ("urn:xccdf:fix:script:sh", "Shell script"),
+    ("urn:xccdf:fix:script:ansible", "Ansible snippet"),
+    ("urn:xccdf:fix:script:puppet", "Puppet snippet"),
+    ("urn:redhat:anaconda:pre", "Anaconda snippet"),
+    ("urn:xccdf:fix:script:kubernetes", "Kubernetes snippet"),
```
Comment (jan-cerny) [inline tests/unit_tests/test_data_structure.py:175]: What about testing also some nonsense like what would happen if eg. "urn:xccdf:fix:script:pejskoskript" be passed?

File: tests/unit_tests/test_data_structure.py [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+        {"low": 0, "medium": 351, "high": 0, "unknown": 0, "sum_of_filed_rules": 351}
+    ),
+    (
+        ("medium", "high", "unknown"),
+        {"low": 33, "medium": 0, "high": 0, "unknown": 0, "sum_of_filed_rules": 33}
+    ),
+])
+def test_report_severity_of_failed_rules_stats(to_remove, result):
+    report = None
+    if "fail" in to_remove:
```
Comment (jan-cerny) [inline tests/unit_tests/test_data_structure.py:154]: This is somehow suspicious condition.
I propose either split this test case into 2 (first with severity and second with test result) or add another parameter `type`.

File: tests/unit_tests/test_data_structure.py [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+            assert result[key] == rule_results_stats[key]
+    else:
+        with pytest.raises(MissingProcessableRules):
+            assert report.get_rule_results_stats()
+
+
+@pytest.mark.parametrize("to_remove, result", [
+    (
+        (),
+        {"low": 33, "medium": 351, "high": 25, "unknown": 33, "sum_of_filed_rules": 442}
```
Comment (jan-cerny) [inline tests/unit_tests/test_data_structure.py:93]: filed or failed?

File: tests/unit_tests/test_scap_result_parser.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    xml_data = None
+    with open(file_path, "r", encoding="utf-8") as xml_report:
+        xml_data = xml_report.read().encode()
+    return SCAPResultsParser(xml_data)
+
+
+@pytest.mark.parametrize("file_path, result", [
+    (PATH_TO_ARF, True),
+    (PATH_TO_XCCDF, False),
+    (PATH_TO_EMPTY_XML_FILE, False),
```
Comment (jan-cerny) [inline tests/unit_tests/test_scap_result_parser.py:20]: And now what about invalid ARF, valid XCCDF, different versions of XCCDF, etc.

File: tests/unit_tests/test_scap_result_parser.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+@pytest.mark.parametrize("file_path, number_of_cpe_platforms", [
+    (PATH_TO_ARF, 13),
+    (PATH_TO_XCCDF, 13),
+])
+def test_get_profile_info(file_path, number_of_cpe_platforms):
+    parser = get_parser(file_path)
+    report = parser.get_profile_info()
+    assert len(report.cpe_platforms) == number_of_cpe_platforms
+    assert report.profile_name is not None
+    assert report.rules is None
```
Comment (jan-cerny) [inline tests/unit_tests/test_scap_result_parser.py:36]: Also test if the actual contents is as expected, not only that it isn't a None.

File: tests/unit_tests/test_scap_result_parser.py [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+
+
+@pytest.mark.parametrize("file_path, number_of_rules", [
+    (PATH_TO_ARF, 714),
+    (PATH_TO_XCCDF, 714),
+])
+def test_get_info_about_rules_in_profile(file_path, number_of_rules):
+    parser = get_parser(file_path)
+    rules = parser.get_info_about_rules_in_profile()
+    assert len(rules.keys()) == number_of_rules
```
Comment (jan-cerny) [inline tests/unit_tests/test_scap_result_parser.py:46]: I miss asserts for the members of the `Rule` class.

File: tests/unit_tests/test_scap_result_parser.py [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+    (PATH_TO_XCCDF, False),
+    (PATH_TO_EMPTY_XML_FILE, False),
+])
+def test_validation(file_path, result):
+    parser = get_parser(file_path)
+    assert parser.validate(parser.arf_schemas_path) == result
+
+
+@pytest.mark.parametrize("file_path, number_of_cpe_platforms", [
+    (PATH_TO_ARF, 13),
```
Comment (jan-cerny) [inline tests/unit_tests/test_scap_result_parser.py:28]: In general, these are some huge XML files. But, for unit tests, I would expect to have many minimal ARF or XCCDF files to have the testing more focused.

File: tests/unit_tests/test_scap_result_parser.py [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+    (PATH_TO_XCCDF, 13),
+])
+def test_get_profile_info(file_path, number_of_cpe_platforms):
+    parser = get_parser(file_path)
+    report = parser.get_profile_info()
+    assert len(report.cpe_platforms) == number_of_cpe_platforms
+    assert report.profile_name is not None
+    assert report.rules is None
+
+
```
Comment (jan-cerny) [inline tests/unit_tests/test_scap_result_parser.py:38]: I miss tests for static methods like _get_full_description, _get_multi_check etc. Preferably on a minimal input data.

**Review by jan-cerny — COMMENTED** [review]
It's a great start. But we should aim for a much larger test coverage.

**Review by jan-cerny — APPROVED** [review]
@Honny1 It's amazing! Good job!
